core/backends/backend-xwiki/src/xwikiStorage.ts

```
@@ -315,6 +316,14 @@ export class XWikiStorage extends AbstractStorage {
       // TODO: the syntax provided by the save is ignored and the content is always saved as markdown.
       body: JSON.stringify({ content, title, syntax: "markdown/1.2" }),
     });
+    if (!response.ok) {
+      const errorMessage = await response.text();
+      // TODO: Fix CRISTAL-383
+      this.alertsServiceProvider
+        .get()
+        .error(`Could not save page ${page}. Reason: ${errorMessage}`);
+      throw new Error(errorMessage);
```

Do we really need to throw an error if it is already handled by the alert service provider?

---

Actually, yes, we need it to update the editor status (that shows "Unsaved" only if it catches an error).

---

ok, maybe add a comment to explain that, so that nobody removes it by accident.

---

Good point.